packages/web/app/src/components/layouts/project-selector.tsx

```diff
@@ -38,47 +47,74 @@ export function ProjectSelector(props: {
 
   return (
     <>
-      {currentOrganization ? (
-        <Link
-          to="/$organizationSlug"
-          params={{ organizationSlug: props.currentOrganizationSlug }}
-          className="max-w-[200px] shrink-0 truncate font-medium"
-        >
-          {currentOrganization.slug}
-        </Link>
+      {showOrganization ? (
+        currentOrganization ? (
+          <Link
+            to="/$organizationSlug"
+            params={{ organizationSlug: props.currentOrganizationSlug }}
+            className="max-w-[200px] shrink-0 truncate font-medium"
+          >
+            {currentOrganization.slug}
+          </Link>
+        ) : (
+          <div className="h-5 w-48 max-w-[200px] animate-pulse rounded-full bg-gray-800" />
+        )
       ) : (
-        <div className="h-5 w-48 max-w-[200px] animate-pulse rounded-full bg-gray-800" />
+        ''
       )}
-      {projects?.length && currentProject ? (
+
+      {(projects?.length && currentProject) || optional ? (
         <>
-          <div className="italic text-gray-500">/</div>
+          {showOrganization ? <div className="italic text-gray-500">/</div> : <></>}
           <Select
             value={props.currentProjectSlug}
-            onValueChange={id => {
-              void router.navigate({
-                to: '/$organizationSlug/$projectSlug',
-                params: {
-                  organizationSlug: props.currentOrganizationSlug,
-                  projectSlug: id,
-                },
-              });
-            }}
+            onValueChange={
+              onValueChangeFunc
+                ? id => {
+                    onValueChangeFunc(id);
```

Comment:
minor: to clean up the jsx, I'd move this defaulting logic to the top of the function alongside the other defaulting logic. And you shouldnt need to wrap the `onValueChangeFunc`